projects/storefrontapp-e2e-cypress/cypress/e2e/vendor/cpq/download-proposal-document.e2e.cy.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as quote from '../../../helpers/vendor/cpq/quote-download';
+
+const BUYER_EMAIL = 'james.weber@harvestlive.inc';
+const BUYER_PASSWORD = 'welcome';
+const BUYER_USER = 'James Weber';
+const MSG_TYPE_WARNING = '[GlobalMessage] Warning';
+
+context('Download proposal document', () => {
+  let globalMessageSettings: any;
+  beforeEach(() => {
+    globalMessageSettings = {
+      globalMessages: {
+        [MSG_TYPE_WARNING]: {
+          timeout: 10000,
+        },
+      },
+    };
+    cy.cxConfig(globalMessageSettings);
+    cy.visit('/');
+    quote.login(BUYER_EMAIL, BUYER_PASSWORD, BUYER_USER);
+    quote.registerReadVendorQuoteRoute();
+    quote.registerReadQuoteRoute();
+    quote.registerDownloadAttachmentRoute();
+  });
+
+  it('should download proposal document', () => {
+    quote.navigateToVendorQuoteListFromMyAccount();
```

Comment:
In your cypress test you relay on the fact that there is a list of the vendor quotes. Normally, end-user has to create a quote, then submit it to a vendor. What I want to say, you can not cut short a process.

Reply:
The process of creating a quote and submitting to a vendor is already covered by other existing cypress tests.
This tests only checks the download functionality, assuming that the above processes are done and vendor has attached a proposal document to the quote.
We are manually generating and sending quotes from CPQ, and focusing only on the download button click in the cypress test.

Reply:
I am aware that the corresponding tests exist.
My point is how you could guaranty that the above processes have been completed when you have not trigger them.
If not, then the quote list is empty and you are not able to download the proposal.

Reply:
For instance, one has to test that the address form in the checkout process has been correctly filled.
To do it, one has to add a product to the cart and proceed to the checkout and then the address form, you are not able to jump directly to the address form of the checkout process or rely that some other test did it before.

Reply:
But then one of the steps that must happen for the download button to be visible depends on an external element CPQ. Only when a vendor generates and sends back a proposal document from CPQ, it is visible in Spartacus UI. 
If we were to cover all the process that happens until the download button is visible and checked upon, how would we replicate the above scenario in the cypress tests?

Reply:
There are 2 CPQ cypress suite: 
1. cpq-configuration.ccv2-e2e.cy.ts
2. cpq-configuration.core-e2e.cy.ts
There is a script `cy:run:cpq` to run them in projects/storefrontapp-e2e-cypress/`package.json` file

Reply:
In `quote.e2e-2211.FP4.cy.ts` file you could find some examples how to create, edit , cancel or submit a quote then how to log in in asm mode and how to submit, cancel or edit a quote as a vendor. 
When you require more APis then you could extend the existing `b2b-quote.ts` file.
Please, don't create a seperate one to click on the `download proposal` button.
This action is for the quote and could be added to the existing file.

Reply:
@Larisa-Staroverova I agree to what @suprishi says: In this case we cannot write fully self-contained tests (as there is no way how we can trigger the proposal generation in CPQ. We have only a couple of test systems where CPQ is integrated at all).
The quote negotiation is not done using the ASM scenario here!
So I fear we need a dedicated suite...

Reply:
So does it mean that the test suite for the download button and its helper API functions remain in the way they currently are?

Reply:
Hi @suprishi, yes, but, please, reuse existing API such as login, logout and soon.
Please, implement methods that are required to download a proposal and reuse existing ones.